packaging/linux/buildpackage.sh

```diff
@@ -0,0 +1,120 @@
+	exit 1
+fi
+
+PACKAGING_DIR=$(python -c "import os; print(os.path.dirname(os.path.realpath('$0')))")
+TEMPLATE_ROOT="${PACKAGING_DIR}/../../"
+
+# shellcheck source=mod.config
+. "${TEMPLATE_ROOT}/mod.config"
+
+if [ -f "${TEMPLATE_ROOT}/user.config" ]; then
+	# shellcheck source=user.config
+	. "${TEMPLATE_ROOT}/user.config"
+fi
+
+if [ "${INCLUDE_DEFAULT_MODS}" = "True" ]; then
+	echo "Cannot generate installers while INCLUDE_DEFAULT_MODS is enabled."
+	echo "Make sure that this setting is disabled in both your mod.config and user.config."
+	exit 1
+fi
+
+TAG="$1"
+if [ $# -eq "1" ]; then
+	OUTPUTDIR=$(python -c "import os; print(os.path.realpath('.'))")
+else
+	OUTPUTDIR=$(python -c "import os; print(os.path.realpath('$2'))")
+fi
+
+BUILTDIR="${PACKAGING_DIR}/${PACKAGING_INSTALLER_NAME}.appdir"
+
+# Set the working dir to the location of this script
+cd "${PACKAGING_DIR}"
+
+pushd ${TEMPLATE_ROOT} > /dev/null
+
+if [ ! -f "${ENGINE_DIRECTORY}/Makefile" ]; then
+	echo "Required engine files not found."
+	echo "Run \`make\` in the mod directory to fetch and build the required files, then try again.";
+	exit 1
+fi
+
+if [ ! -d "${OUTPUTDIR}" ]; then
+	echo "Output directory '${OUTPUTDIR}' does not exist.";
+	exit 1
+fi
+
+echo "Building core files"
+
+make version VERSION="${TAG}"
+
+pushd ${ENGINE_DIRECTORY} > /dev/null
+make linux-dependencies
+make core SDK="-sdk:4.5"
+make install-engine prefix="usr" DESTDIR="${BUILTDIR}/"
+make install-common-mod-files prefix="usr" DESTDIR="${BUILTDIR}/"
+
+popd > /dev/null
+popd > /dev/null
+
+# Add native libraries
+echo "Downloading dependencies"
+curl -s -L -o "${PACKAGING_APPIMAGE_DEPENDENCIES_TEMP_ARCHIVE_NAME}" -O "${PACKAGING_APPIMAGE_DEPENDENCIES_SOURCE}" || exit 3
+tar xf "${PACKAGING_APPIMAGE_DEPENDENCIES_TEMP_ARCHIVE_NAME}"
+
+curl -s -L -O https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage
+chmod a+x appimagetool-x86_64.AppImage
+
+echo "Building AppImage"
+
+# Add mod files
+cp -r "${TEMPLATE_ROOT}/mods/"* "${BUILTDIR}/usr/lib/openra/mods"
+
+install -Dm 0755 libSDL2.so "${BUILTDIR}/usr/lib/openra/"
+install -Dm 0644 include/SDL2-CS.dll.config "${BUILTDIR}/usr/lib/openra/"
+install -Dm 0755 libopenal.so.1 "${BUILTDIR}/usr/lib/openra/"
+install -Dm 0644 include/OpenAL-CS.dll.config "${BUILTDIR}/usr/lib/openra/"
+install -Dm 0755 liblua.so.5.1 "${BUILTDIR}/usr/lib/openra/"
+install -Dm 0644 include/Eluant.dll.config "${BUILTDIR}/usr/lib/openra/"
+
+# Add launcher and icons
+sed "s/{MODID}/${MOD_ID}/g" include/AppRun.in > AppRun.temp
+install -m 0755 AppRun.temp "${BUILTDIR}/AppRun"
+
+sed "s/{MODID}/${MOD_ID}/g" include/mod.desktop.in | sed "s/{MODNAME}/${PACKAGING_DISPLAY_NAME}/g" > temp.desktop
+install -Dm 0755 temp.desktop "${BUILTDIR}/usr/share/applications/openra-${MOD_ID}.desktop"
+
+install -m 0755 temp.desktop "${BUILTDIR}/openra-${MOD_ID}.desktop"
+install -m 0644 mod_128x128.png "${BUILTDIR}/openra-${MOD_ID}.png"
```

Comment:
Hardcoding this is bad.  This should move into the `for` below so that it will copy the largest available image.  I should also include at least a 256x256 sized icon by default.